pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -421,33 +421,42 @@ func (d *namespacedResourcesDeleter) deleteEachItem(gvr schema.GroupVersionResou
 	return nil
 }
 
+type gvrDeletionMetadata struct {
+	// finalizerEstimateSeconds is an estimate of how much longer to wait
+	finalizerEstimateSeconds int64
+	// numRemaining is how many instances of the gvr remain
```

Comment:
comment: zero means no estimate has been made, not that we're done (or something like this)